action/protocol/execution/evm/evm.go

```diff
@@ -145,6 +145,9 @@ func ExecuteContract(
 		return nil, err
 	}
 	retval, depositGas, remainingGas, contractAddress, err := executeInEVM(ps, stateDB, raCtx.GasLimit)
+	if errors.Cause(err) == action.ErrHitGasLimit {
```

Comment:
should we also early exit on action.ErrOutOfGas?
executeInEVM() returns ErrOutOfGas on line 200

Reply:
Unlike ErrHitGasLimit, ErrOutOfGas is the error of the execution itself, so we should handle it uniformly as other error cases.